src/test/java/com/scaleunlimited/flinkcrawler/topology/CrawlTopologyTest.java

```diff
@@ -43,16 +49,117 @@
+                                        new MockUrlLengthenerFetcher())))
+                .setRobotsFetcherBuilder(
+                        new MockRobotsFetcher.MockRobotsFetcherBuilder(new MockRobotsFetcher()))
+                .setRobotsParser(new SimpleRobotRulesParser())
+                .setPageParser(new SimplePageParser(new ParserPolicy(), new PageNumberScorer()))
+                .setContentSink(new DiscardingSink<ParsedUrl>())
+                .setContentTextFile(contentTextFile.getAbsolutePath())
+                .setUrlNormalizer(normalizer)
+                .setUrlFilter(new SimpleUrlValidator())
+                .setSiteMapFetcherBuilder(
+                        new SiteMapGraphFetcher.SiteMapGraphFetcherBuilder(new SiteMapGraphFetcher(
+                                BaseWebGraph.EMPTY_GRAPH)))
+                .setSiteMapParser(new SimpleSiteMapParser())
+                .setDefaultCrawlDelay(0)
+                .setPageFetcherBuilder(
+                        new WebGraphFetcher.WebGraphFetcherBuilder(new WebGraphFetcher(graph)))
+                .setFetchQueue(new FetchQueue(10_000, minFetchScore));
+
+        CrawlTopology ct = builder.build();
+
+        File dotFile = new File(testDir, "topology.dot");
+        ct.printDotFile(dotFile);
+
+        // Execute for a maximum of 20 seconds.
+        ct.execute(20_000);
+
+        for (Tuple3<Class<?>, String, Map<String, String>> entry : UrlLogger.getLog()) {
+            LOGGER.debug("{}: {}", entry.f0, entry.f1);
+        }
+
+        String domain1page1 = normalizer.normalize("domain1.com/page1");
+        String domain1page2 = normalizer.normalize("domain1.com/page2");
+        String domain1page3 = normalizer.normalize("domain1.com/page3");
+        String domain1page4 = normalizer.normalize("domain1.com/page4");
+        String domain1page5 = normalizer.normalize("domain1.com/page5");
+
+        UrlLoggerResults results = new UrlLoggerResults(UrlLogger.getLog());
+
+        results.assertUrlLoggedBy(FetchUrlsFunction.class, domain1page1, 1)
+                .assertUrlLoggedBy(FetchUrlsFunction.class, domain1page2, 1)
+                // This page never got a high enough estimated score.
+                .assertUrlLoggedBy(FetchUrlsFunction.class, domain1page3, 0)
+                .assertUrlLoggedBy(FetchUrlsFunction.class, domain1page4, 1)
+                .assertUrlLoggedBy(FetchUrlsFunction.class, domain1page5, 1);
+    }
+
     @SuppressWarnings("deprecation")
     @Test
-    public void testAsync() throws Exception {
+    public void testBroadCrawl() throws Exception {
         UrlLogger.clear();
 
         LocalStreamEnvironmentWithAsyncExecution env = new LocalStreamEnvironmentWithAsyncExecution();
 
-        // Set up for checkpointing with in-memory state. Our test runs for several seconds, so using
+        // Set up for checkpointing with in-memory state. Our test runs for several seconds, so
+        // using
```

Comment:
The reformatting of this file (and others) seems to have mucked up comments a bit. Plus the lines below have different indentation, which makes me wonder if they still are lined up properly when we're doing `.blah()` and then a `.blah()` on a following line.

Reply:
Yes, as I was rushing to put together this PR I applied Source>Format to my entire flink-crawler directory from within Eclipse. I noted with dismay that it had mucked around with a ton of files. I laboriously tried to back out all of the changes to files I hadn't modified myself, but I remember one that was in the same hunk as a change I did make (probably the one above), and I didn't have time to fix that up.

Looking through the other (mostly indentation) format changes now, I recognize similar results (to what I think I saw the last time I tried to reformat from within Eclipse?) I was trying to remember what I was doing last time and note that [your reformatting instructions](https://github.com/ScaleUnlimited/flink-crawler/pull/145) use the java-formatter Maven plugin. I think I remember trying a different plugin for some reason last time, but I don't remember why (trying to automate the commit process? Eclipse version compatibility?) :(

Perhaps code (re-)formatting instructions (and the Scale Unlimited profile) should be added to the [Best Practices](https://github.com/ScaleUnlimited/flink-crawler/wiki/Best-Practices) wiki page to clarify the procedure for committing properly formatted code.